pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -157,6 +165,18 @@ func (pc *PodCache) onEvent(curr interface{}, ev model.Event) error {
 			} else if shouldPodBeInEndpoints(pod) {
 				if key != pc.podsByIP[ip] {
 					pc.update(ip, key)
+				} else if !model.WorkloadInstancesEqual(pc.c.workloadInstancesByPods[key], instance) {
+					// fire off eds update on pod change
+					svcList, err := pc.c.serviceLister.Services(pod.Namespace).List(k8sLabels.Everything())
```

Comment:
Currently the pod labels for a connected pod are static as they are part of the XDS node metadata which does not update.

When changing this, are we only solving one end of the problem that may lead to unexpected state?

Reply:
Do you mean we should update XDS node metadata when pod label changes?

Reply:
We do not consider changing pod labels as a good practice, and so did not make it at first. I doubt it is not worth to support it with so much performance sacrifice.

Reply:
I see, but it sounds a bit odd that the rule would be different after istiod restarted

Generally pod update events(**which we care**) do not happen frequently, does it really kill performance a lot?
And actually `ServiceEntryStore.WorkloadInstanceHandler` does similar things as this patch:
https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/serviceentry/servicediscovery.go#L441-L468

Anyway I'm ok if we really don't need this feature

Reply:
> Do you mean we should update XDS node metadata when pod label changes?

AFAIK this requires restarting Envoy which is not really viable

Reply:
> AFAIK this requires restarting Envoy which is not really viable

Yeah, it's bootstrap configuration

Anyway I think we should "keep the same behavior as workload handler" which does [fire eds update on workload change](https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/serviceentry/servicediscovery.go#L468), WDYT?

Reply:
Or maybe give an env variable(default to false) to control this?